File: rdf4j/src/main/scala/eu/ostrzyciel/jelly/convert/rdf4j/rio/JellyWriterFactory.scala
package eu.ostrzyciel.jelly.convert.rdf4j.rio

import org.eclipse.rdf4j.rio.{RDFFormat, RDFWriter, RDFWriterFactory}

import java.io.{OutputStream, Writer}

final class JellyWriterFactory extends RDFWriterFactory:
  override def getRDFFormat: RDFFormat = JELLY

  override def getWriter(out: OutputStream): JellyWriter = JellyWriter(out)

  override def getWriter(out: OutputStream, baseURI: String): JellyWriter = getWriter(out)

  override def getWriter(writer: Writer): RDFWriter = throw new UnsupportedOperationException

  override def getWriter(writer: Writer, baseURI: String): RDFWriter = throw new UnsupportedOperationException

File: rdf4j/src/main/java/eu/neverblink/jelly/convert/rdf4j/rio/JellyWriterFactory.java
package eu.neverblink.jelly.convert.rdf4j.rio;

import static eu.neverblink.jelly.convert.rdf4j.rio.JellyFormat.JELLY;

import eu.neverblink.jelly.convert.rdf4j.Rdf4jConverterFactory;
import java.io.OutputStream;
import java.io.Writer;
import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
import org.eclipse.rdf4j.rio.RDFFormat;
import org.eclipse.rdf4j.rio.RDFWriter;
import org.eclipse.rdf4j.rio.RDFWriterFactory;

public final class JellyWriterFactory implements RDFWriterFactory {

    @Override
    public RDFFormat getRDFFormat() {
        return JELLY;
    }

    @Override
    public RDFWriter getWriter(OutputStream out) {
        final var converterFactory = Rdf4jConverterFactory.getInstance();
        final var valueFactory = SimpleValueFactory.getInstance();
        return new JellyWriter(converterFactory, valueFactory, out);
    }

    @Override
    public RDFWriter getWriter(OutputStream out, String baseURI) {
        return getWriter(out);
    }

    @Override
    public RDFWriter getWriter(Writer writer) {
        throw new UnsupportedOperationException();
    }

    @Override
    public RDFWriter getWriter(Writer writer, String baseURI) {
        throw new UnsupportedOperationException();
    }
}
